test/fixtures/tls-connect.js

```diff
@@ -92,7 +92,7 @@ exports.connect = function connect(options, callback) {
   function maybeCallback() {
     if (!callback)
       return;
-    if (server.conn && (client.conn || client.err)) {
```

Comment:
`client.err` is set by the `'error'` listener of the client socket or in the `catch` branch of the `try...catch` around `tls.connect()`. In the latter case `maybeCallback()` is not used ad the callback is called directly. In the former `client.conn` is also set.